docs/modules/ROOT/pages/reference/extensions/jfr.adoc

```diff
@@ -34,38 +38,75 @@ Please refer to the above link for usage and configuration details.
 
 Check the xref:user-guide/index.adoc[User guide] for more information about writing Camel Quarkus applications.
 
+== Usage
+
+### Starting flight recording
+
+To enable Java Flight Recorder to start a recording and automatically dump the recording to disk after Camel startup is complete, use the following configuration in `application.properties`.
+
+[source,properties]
+----
+quarkus.camel.jfr.startup-recorder-recording=true
+----
+
+Alternatively you can pass some Java options to the runnable application JAR or the native executable to enable flight recording at application startup.
+
+In JVM mode the application runnable JAR can be executed as follows.
+
+[source,shell]
+----
+$ java -XX:+FlightRecorder -XX:StartFlightRecording=filename=recording.jfr -jar quarkus-run.jar
+----
+
+In native mode, the native executable can be executed as follows.
+
+[source,shell]
+----
+$ ./my-application-runner -XX:+FlightRecorder -XX:StartFlightRecording=filename=recording.jfr
+----
+
+### Flight recording in native mode
+
+To build a native image with Java Flight Recorder support, it is mandatory to enable VM inspection via the following configuration property in `application.properties`.
+
+[source,properties]
+----
+quarkus.native.enable-vm-inspection=true
+----
+
+
 == Additional Camel Quarkus configuration
 
 [width="100%",cols="80,5,15",options="header"]
 |===
 | Configuration property | Type | Default
 
 
```

Comment:
It's not really linked to this PR. I've noticed that the default value column is empty.

Reply:
It's because I did not specify any default values on the config items. Camel has its own internal defaults which I mention in the config property documentation.

Maybe I should specify the defaults in `RuntimeCamelJfrConfig` so that it's clear in the docs?

Reply:
Having the table filled with the default value would looks clearer from doc perspective. However, I read it like the true default values come from Camel, so maybe we should not maintain that in camel-quakus ?
I don't know if it's possible to point to camel default values.

Reply:
I did some experiments and have decided to leave things as-is and not duplicate the defaults that are already defined in Camel.

I may follow up with another PR later to refactor the way the JFR config works to simplify it a bit.